src/shacl/ShaclParser.ts

```diff
@@ -0,0 +1,583 @@
+        },
+        {
+          ALT: () => this.SUBRULE(this.shaclTargetNode),
+        },
+        {
+          ALT: () => this.SUBRULE(this.shaclPropertyPath),
+        },
+        {
+          ALT: () => this.SUBRULE(this.shaclLiteralConstraint),
+        },
+        {
+          ALT: () => this.SUBRULE(this.shaclListTakingConstraint),
+        },
+        {
+          ALT: () => this.SUBRULE(this.shaclShapeExpectingConstraint),
+        },
+        {
+          ALT: () => this.SUBRULE(this.shaclHasValueConstraint),
+        },
+        {
+          ALT: () => this.SUBRULE(this.shaclVerbShape),
+        },
+      ]);
+    }
+  );
+
+  shaclPredicateIRI = this.RULE('shaclPredicateIRI', () => {
+    this.CONSUME(categoryTokenMap.IriTakingPredicate);
+    this.SUBRULE(this.iri); // FIXME? This may need to allow a list of IRIs, as with shaclTargetNode.
+  });
+
+  shaclNodeKind = this.RULE('shaclNodeKind', () => {
+    this.CONSUME(this.shaclTokenMap.SHACL_nodeKind);
+    this.CONSUME(categoryTokenMap.NodeKindIRI);
+  });
+
+  shaclTargetNode = this.RULE('shaclTargetNode', () => {
+    this.CONSUME(this.shaclTokenMap.SHACL_targetNode);
+    this.SUBRULE(this.shaclIRIOrLiteral);
+    this.MANY(() => {
+      this.CONSUME(turtleTokenMap.Comma);
+      this.SUBRULE1(this.shaclIRIOrLiteral);
+    });
+  });
+
+  shaclVerbShape = this.RULE('shaclVerbShape', () => {
+    this.SUBRULE(this.verb);
+    this.SUBRULE(this.shaclShapeType);
+    this.MANY(() => {
+      this.CONSUME(turtleTokenMap.Comma);
+      this.SUBRULE1(this.shaclShapeType);
+    });
+  });
+
+  shaclShapeType = this.RULE('shaclShapeType', () => {
+    this.OR([
+      {
+        ALT: () => this.CONSUME(this.shaclTokenMap.SHACL_Shape),
+      },
+      {
+        ALT: () => this.CONSUME(this.shaclTokenMap.SHACL_NodeShape),
+      },
+      {
+        ALT: () => this.CONSUME(this.shaclTokenMap.SHACL_PropertyShape),
+      },
+    ]);
+  });
+
+  shaclPropertyPath = this.RULE('shaclPropertyPath', () => {
+    this.CONSUME(this.shaclTokenMap.SHACL_path);
+    this.SUBRULE(this.shaclPropertyPathPath);
+  });
+
+  shaclPropertyPathPath = this.RULE('shaclPropertyPathPath', () => {
+    this.OR([
+      {
+        ALT: () => this.SUBRULE(this.shaclPredicatePath),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclSequencePath),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclAlternativePath),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclInversePath),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclZeroOrMorePath),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclOneOrMorePath),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclZeroOrOnePath),
+      },
+    ]);
+  });
+
+  shaclPredicatePath = this.RULE('shaclPredicatePath', () => {
+    this.OR([
+      {
+        ALT: () => this.SUBRULE(this.iri),
+      },
+      {
+        // This case does not seem to be allowed by the SHACL spec, but the
```

Comment:
ah, it's not a `Sequence Path` b/c that specifically requires at least two members..
https://www.w3.org/TR/shacl/#property-path-sequence